tasks.py

```diff
@@ -245,100 +245,86 @@ def load_osm_streets(ctx, files=[]):
-                --connection-string {ctx.es} \
-                {additional_params} \
-                --dataset {ctx.dataset}".format(
-                    ctx=ctx, additional_params=additional_params
-                ),
-            )
-        if ctx.addresses.get("oa", {}).get("path"):
-            logging.info("importing oa addresses")
-            conf = ctx.addresses.oa
-            additional_params = _get_cli_param(conf.get("nb_threads"), "--nb-threads")
-            additional_params += _get_cli_param(conf.get("nb_shards"), "--nb-shards")
-            additional_params += _get_cli_param(conf.get("nb_replicas"), "--nb-replicas")
-            run_rust_binary(
-                ctx,
-                "mimir",
-                "openaddresses2mimir",
-                files,
-                "--input {ctx.addresses.oa.path} \
-                --connection-string {ctx.es} \
-                {additional_params} \
-                --dataset {ctx.dataset}".format(
-                    ctx=ctx, additional_params=additional_params
-                ),
-            )
-        return
 
-    # -- Use deduplication
+def load_addresses_base_params(ctx):
+    conf = ctx.addresses.bano
+    return [
+        _get_cli_param(conf.get("nb_threads"), "--nb-threads"),
+        _get_cli_param(conf.get("nb_shards"), "--nb-shards"),
+        _get_cli_param(conf.get("nb_replicas"), "--nb-replicas"),
+        _get_cli_param(ctx.es, "--connection-string"),
+        _get_cli_param(ctx.dataset, "--dataset"),
+    ]
 
-    output_csv = "/data/addresses/output.csv.gz"
 
-    if ctx.addresses.get("skip_deduplication") and file_exists(ctx, files, output_csv):
-        logging.info("`%s` already exists: skipping deduplication", output_csv)
+@task()
+def load_bano_adresses(ctx, input_file, files=[]):
+    """Populate ES with addresses from a BANO file"""
+    logging.info("importing bano addresses from %s", input_file)
+    params = load_addresses_base_params(ctx) + [_get_cli_param(input_file, "--input")]
+    run_rust_binary(ctx, "mimir", "bano2mimir", files, " ".join(params))
+
+
+@task()
+def load_oa_addresses(ctx, input_path, files=[]):
+    """Populate ES with addresses from an OpenAddresses file"""
+    logging.info("importing oa addresses from %s", input_path)
+    params = load_addresses_base_params(ctx) + [_get_cli_param(input_path, "--input")]
+    run_rust_binary(ctx, "mimir", "openaddresses2mimir", files, " ".join(params))
+
+
+@task()
+def load_addresses(ctx, files=[]):
+    """Fetch addresses and populate the database"""
+    if not _is_config_object(ctx.get("addresses")):
+        logging.info("no addresses to import")
         return
+
+    if ctx.addresses.deduplication.enable:
+        output_csv = ctx.addresses.deduplication.output
+
+        if ctx.addresses.deduplication.skip and file_exists(ctx, files, output_csv):
+            logging.info("`%s` already exists: skipping deduplication", output_csv)
+            return
```

Comment:
It's still a bit confusing to me that `deduplication.skip = true` will not only skip the dedupe task, but also the import to Elasticsearch. In particular, any update in the "admins" dataset that would have been loaded in the meantime, will not be reflected in the addresses index.

Reply:
Oh you're right, it would make more sense that it only affects the deduplication, plus I think it may be more helpful